src/js/letters/containers/AddressSection.jsx

```diff
@@ -39,31 +51,28 @@ export class AddressSection extends React.Component {
 
   handleChange = (path, update) => {
     this.setState(({ editableAddress }) => {
-      // reset state code when user changes address country but don't add or
-      // modify state property otherwise
-      return (path === 'country'
+      // reset state code when user updates country
+      const newFragment = (path === 'country'
         ? {
-          editableAddress: {
-            ...editableAddress,
-            [path]: update,
-            state: '',
-            militaryStateCode: '',
-          },
-        }
-        : {
-          editableAddress: {
-            ...editableAddress,
-            [path]: update,
-          },
+          [path]: update,
+          state: '',
+          militaryStateCode: '',
```

Comment:
I don't want to block this merge, so I'm tracking this [here](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/5015): We don't need this `militaryStateCode`.